### PR TITLE
Parent object addition/removal

### PR DESCRIPTION
**Pull request by paul-turner**
Adds ability to add/remove parent objects into the adapter.

### CODE REVIEW

**Comment by DarekDeo** [comment]
Looks ok, although it allows to add only 1 object at a time. For example: while asking some external API for items, API will retrieve more likely list of items.

**Comment by paul-turner** [comment]
Fixed master conflicts.

Will work on adding list of items in a subsequent PR.

File: app/src/main/res/layout/activity_horizontal_linear_recycler_view_sample.xml [hunk 1]
```diff
@@ -52,4 +52,44 @@
...
+        android:layout_height="wrap_content"
+        android:text="@string/add_item_to_end"/>
+
+    <Button
+        android:id="@+id/activity_horizontal_linear_recycler_view_remove_from_end_button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/remove_item_from_end"/>
+
+</LinearLayout>
```
Comment (dgreenhalgh) [inline app/src/main/res/layout/activity_horizontal_linear_recycler_view_sample.xml]: Weird tabbing here

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
     private static final int TYPE_CHILD = 1;
 
     protected List<Object> mItemList;
-    protected List<? extends ParentListItem> mParentItemList;
+    protected List<ParentListItem> mParentItemList;
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: ?

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 3]
```diff
@@ -412,6 +414,72 @@ private int getExpandedItemCount(int position) {
...
+
+        int sizeChanged = 1;
+        int wrapperIndex = getParentWrapperIndex(parentPosition);
+        if (wrapperIndex == -1) {
+            throw new IllegalStateException("Parent not found");
+        }
+
+        ParentWrapper parentWrapper = (ParentWrapper) mItemList.remove(wrapperIndex);
+        if (parentWrapper.isExpanded()) {
+            for (int i = 0; i < parentListItem.getChildItemList().size(); i++) {
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: Could pull the `List#size` call out of the loop

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java [hunk 4]
```diff
@@ -412,6 +414,72 @@ private int getExpandedItemCount(int position) {
         return expandedCount;
     }
 
+    // endregion
+
+    // region Data Manipulation
```
Comment (dgreenhalgh) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: This region could use docs

File: expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java (hunk 2 above)
Comment (paul-turner) [inline expandablerecyclerview/src/main/java/com/bignerdranch/expandablerecyclerview/Adapter/ExpandableRecyclerAdapter.java]: a List of `List<? extends ParentListItem>` was not allowing the following code:
```java
    public void addParent(int location, ParentListItem parentListItem) {
        mParentItemList.add(location, parentListItem);
    }
```

**Comment by paul-turner** [comment]
fixed comments

**Comment by dgreenhalgh** [comment]
Looks good :+1:

**Comment by paul-turner** [comment]
Just a note that the library is going a different direction with  #83

**Pull request merged**